pkg/build/generator/generator.go

```diff
@@ -337,6 +338,15 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 		},
 	}
 
+	if binary != nil {
+		build.Spec.Source.Git = nil
+		build.Spec.Source.Binary = binary
+		build.Spec.Source.Type = buildapi.BuildSourceBinary
+		if build.Spec.Source.Dockerfile != nil && binary.AsFile == "Dockerfile" {
```

Comment:
I was approaching this from user perspective - if I say "start-build --from-file=Dockerfile" I expect dockerfile to take precedence.  If I don't pass that, I'm probably passing a source repo, so if I have a Dockerfile on the build config, it's probably necessary for some reason.  The former scenario has to work, the latter I was ambivalent on.